### ISSUE TITLE
Speech Commands Dataset: Test set different from the Set given in testing_list.txt

### ISSUE THREAD

**Issue by jadeaffenjaeger** [tensorflow/datasets#3061]
**What I need help with / What I was wondering**
For the Google Speech Commands Dataset, it seems to be common practice to derive the test set from the file `testing_list.txt` found in the `speech_commands_v0.02.tar.gz` archive. This file contains 11005 filenames, drawn from the original distribution of 20 words with ca. 400 samples each and no separate class for background noise.

Different from that, Tensorflow Datasets generates the test set from an archive that is downloaded separately:
https://github.com/tensorflow/datasets/blob/34aeabf177a0723d14a902e9cc9022e7a16ab6fc/tensorflow_datasets/audio/speech_commands.py#L54

Inspecting the test split generated by TFDS shows that it contains 4890 test samples with around 400 samples for the 10 target words along with an equal amount of samples for the *background noise* and *unknown* classes. The original testing list seems to be parsed and excluded from the training set, but isn't used otherwise
https://github.com/tensorflow/datasets/blob/34aeabf177a0723d14a902e9cc9022e7a16ab6fc/tensorflow_datasets/audio/speech_commands.py#L166-L168

This means that the test set used is significantly different from the training/validation sets because it's not skewed towards the *_unknown_* class. This bias is also present in the test set generated from the `testing_list.txt`, but not here. I understand how these balanced classes are more useful for assessing model performance, however it makes the performance also much harder to compare to the numbers found in literature.

Is there any other reasoning behind using the different test set?
Is there a simple way of changing the default behavior to use the samples given in the testing_list file?

**It would be nice if...**
There was a hint regarding the difference in test sets on the [description page](https://www.tensorflow.org/datasets/catalog/speech_commands). Ideally even a switch that allows a user to select which one they would want to use.

**Comment by jadeaffenjaeger** [comment]
I just checked against the [original paper](https://arxiv.org/pdf/1804.03209.pdf). The dataset downloaded by TFDS is the one that is supposed to be used for obtaining a Top-One Metric (Section 7.1), while the files given in `testing_list.txt` are supposed to be used for general model evaluation (section 7).

The latter one seems to be a more sensible default to me, though it would be nice if both options were present!
